autoscale/src/main/java/com/sequenceiq/periscope/service/NodeDeletionService.java

```diff
@@ -0,0 +1,86 @@
+
+import com.google.common.collect.Lists;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.common.Status;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.response.StackV4Response;
+import com.sequenceiq.cloudbreak.message.CloudbreakMessagesService;
+import com.sequenceiq.periscope.api.model.ScalingStatus;
+import com.sequenceiq.periscope.common.MessageCode;
+import com.sequenceiq.periscope.domain.Cluster;
+import com.sequenceiq.periscope.domain.LoadAlert;
+import com.sequenceiq.periscope.monitor.handler.CloudbreakCommunicator;
+import com.sequenceiq.periscope.notification.HttpNotificationSender;
+import com.sequenceiq.periscope.utils.StackResponseUtils;
+
+@Service
+public class NodeDeletionService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NodeDeletionService.class);
+
+    @Inject
+    private CloudbreakCommunicator cloudbreakCommunicator;
+
+    @Inject
+    private StackResponseUtils stackResponseUtils;
+
+    @Inject
+    private HistoryService historyService;
+
+    @Inject
+    private HttpNotificationSender notificationSender;
+
+    @Inject
+    private CloudbreakMessagesService messagesService;
+
+    public void deleteStoppedNodesIfPresent(Cluster cluster) {
+        Set<LoadAlert> loadAlerts = cluster.getLoadAlerts();
+        if (loadAlerts.isEmpty()) {
+            LOGGER.info("No load-based policy present for cluster: {}, skipping request to delete stopped nodes", cluster.getStackCrn());
+            return;
+        }
+
+        StackV4Response stackResponse = cloudbreakCommunicator.getByCrn(cluster.getStackCrn());
+        LoadAlert loadAlert = loadAlerts.iterator().next();
+        String hostGroup = loadAlert.getScalingPolicy().getHostGroup();
+        List<String> stoppedHostIds = stackResponseUtils.getStoppedCloudInstanceIdsInHostGroup(stackResponse, hostGroup);
+
+        if (stoppedHostIds.isEmpty()) {
+            LOGGER.info("No stopped nodes found on cluster: {}, skipping request to delete stopped nodes", cluster.getStackCrn());
+        } else {
+            deleteStoppedNodesForCluster(cluster, hostGroup, stackResponse, stoppedHostIds);
+        }
+    }
+
+    private void deleteStoppedNodesForCluster(Cluster cluster, String hostGroup, StackV4Response stackResponse, List<String> stoppedHostIds) {
+        try {
+            LOGGER.info("Sending request to delete stopped nodes with hostIds: {} for cluster: {}", stoppedHostIds, cluster.getStackCrn());
+            cloudbreakCommunicator.deleteInstancesForCluster(cluster, stoppedHostIds);
+            createAutoscalingStoppedNodeDeletionHistoryAndNotify(cluster, hostGroup, stoppedHostIds);
+        } catch (Exception e) {
+            LOGGER.error("Error when trying to delete stopped nodes with hostIds: {} for cluster: {}", stoppedHostIds, cluster.getStackCrn(), e);
+            createAutoscalingStoppedNodeDeletionFailedHistoryAndNotify(cluster, stackResponse.getStatus(), stoppedHostIds);
```

Comment:
Is there a more useful message that this API provides which could be logged to the console? (Typically CB APIs have a concise FailureReason of some kind)

Reply:
Updated this to parse the exception relayed by CB and propagating it to the activity console